Formula/libosmium.rb

```diff
@@ -0,0 +1,21 @@
+class Libosmium < Formula
+  desc "Fast and flexible C++ library for working with OpenStreetMap data."
+  homepage "http://osmcode.org/libosmium/"
+  url "https://github.com/osmcode/libosmium/archive/v2.9.0.tar.gz"
+  sha256 "d4312c0cd9cca41fbf2927aee67cbb95d566540a76bb01fc643e8ba76db13419"
+
+  depends_on "cmake" => :build
+  depends_on "boost" => :build
+  depends_on "google-sparsehash" => :optional
+  depends_on "expat" => :optional
+  depends_on "gdal" => :optional
+  depends_on "proj" => :optional
+  depends_on "doxygen" => :optional
```

Comment:
libosmium itself is rather versatile, so if users are doing some slightly sophisticated geo transformations, they will need gdal/proj.
Expat is for additional file format parsing capabilities, so I would say it's rather useful.

Doxygen can generate documentation, not the most necessary thing, but can be useful since it's a library.